app/src/pages/inside/testsPage/testsPage.jsx

```diff
@@ -42,244 +42,208 @@ import {
   loadingSelector,
 } from 'controllers/testItem';
-import { prevTestItemSelector } from 'controllers/pages';
+import { prevTestItemSelector, userRolesSelector } from 'controllers/pages';
 import { ENTITY_START_TIME } from 'components/filterEntities/constants';
+import { canManageTestItemsActions } from 'common/utils/permissions/permissions';
 
-@connect(
-  (state) => ({
-    debugMode: debugModeSelector(state),
-    validationErrors: validationErrorsSelector(state),
-    tests: testsSelector(state),
-    selectedTests: selectedTestsSelector(state),
-    parentItem: parentItemSelector(state),
-    loading: loadingSelector(state),
-    highlightItemId: prevTestItemSelector(state),
-  }),
-  {
-    toggleTestSelectionAction,
-    unselectAllTestsAction,
-    toggleAllTestsAction,
-    selectTestsAction,
-    fetchTestItemsAction,
-  },
-)
-@withSortingURL({
-  defaultFields: [ENTITY_START_TIME],
-  defaultDirection: SORTING_ASC,
-  namespaceSelector,
-})
-@withPagination({
-  paginationSelector: testPaginationSelector,
-  namespaceSelector,
-})
-@track()
-export class TestsPage extends Component {
-  static propTypes = {
-    deleteItems: PropTypes.func,
-    onEditItem: PropTypes.func,
-    onEditItems: PropTypes.func,
-    validationErrors: PropTypes.object.isRequired,
-    debugMode: PropTypes.bool.isRequired,
-    tests: PropTypes.arrayOf(PropTypes.object),
-    selectedTests: PropTypes.arrayOf(PropTypes.object),
-    activePage: PropTypes.number,
-    itemCount: PropTypes.number,
-    pageCount: PropTypes.number,
-    pageSize: PropTypes.number,
-    sortingColumn: PropTypes.string,
-    sortingDirection: PropTypes.string,
-    fetchTestItemsAction: PropTypes.func,
-    onChangePage: PropTypes.func,
-    onChangePageSize: PropTypes.func,
-    onChangeSorting: PropTypes.func,
-    toggleTestSelectionAction: PropTypes.func,
-    unselectAllTestsAction: PropTypes.func,
-    toggleAllTestsAction: PropTypes.func,
-    selectTestsAction: PropTypes.func,
-    parentItem: PropTypes.object,
-    loading: PropTypes.bool,
-    onFilterAdd: PropTypes.func,
-    onFilterRemove: PropTypes.func,
-    onFilterValidate: PropTypes.func,
-    onFilterChange: PropTypes.func,
-    filterErrors: PropTypes.object,
-    filterEntities: PropTypes.array,
-    tracking: PropTypes.shape({
-      trackEvent: PropTypes.func,
-      getTrackingData: PropTypes.func,
-    }).isRequired,
-    highlightItemId: PropTypes.number,
-  };
+export const TestsPageWrapped = ({
+  deleteItems,
+  onEditItem,
+  onEditItems,
+  activePage,
+  itemCount,
+  pageCount,
+  pageSize,
+  sortingColumn,
+  sortingDirection,
+  onChangePage,
+  onChangePageSize,
+  onChangeSorting,
+  onFilterAdd,
+  onFilterRemove,
+  onFilterValidate,
+  onFilterChange,
+  filterErrors,
+  filterEntities,
+}) => {
+  const { trackEvent } = useTracking();
+
+  const [highlightedRowId, setHighlightedRowId] = useState(null);
+  const [isGridRowHighlighted, setIsGridRowHighlighted] = useState(false);
```

Comment:
Could you please leave a TODO comment to think about moving the highlighting boilerplate logic to a reusable custom hook.
I suppose it would be useful.